src/utils.js

```diff
@@ -0,0 +1,48 @@
+const { chunk } = require('lodash')
+const { findSavedTripByDates } = require('./queries')
+const { differenceWith } = require('lodash')
+
+function canSaveNextTripsChunk(startExecTime, timeout) {
+  const executionTimeSeconds = (new Date() - startExecTime) / 1000
+  return executionTimeSeconds < timeout
+}
+
+async function restartKonnector(client) {
+  const args = {
+    slug: 'tracemob'
+  }
+  const jobCollection = client.collection('io.cozy.jobs')
+  return jobCollection.create('konnector', args)
+}
+
+function createChunks(tripsMetadata, chunkSize) {
+  return chunk(tripsMetadata, chunkSize)
+}
+
+async function keepOnlyNewTrips(trips, accountId) {
+  const firstDate = trips[0].properties.start_fmt_time
```

Comment:
What if the user has no trip? Not sure if code can execute until there if no trip is retrieved in run.js but I'm not sure I read the code correctly

Reply:
This method will never be called if there is no trip. But we should check it here nonetheless :+1: